File: trunk/net/NetTupleEngine/AtomToken.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace it.mintlab.desktopnet.tupleengine
{
    class AtomToken : IToken
    {
        private string atom;
        private object parameter;
        private bool withParameter;

        public AtomToken(string atom)
        {
            this.atom = atom;
            this.withParameter = false;
            this.parameter = null;
        }

        public void setParameter(object parameter)
        {
            this.parameter = parameter;
            this.withParameter = true;
        }

        public object getParameter()
        {
            return parameter;
        }

        public override string ToString()
        {
            return atom;
        }

        public bool isWithParameter() 
        {
            return this.withParameter;
        }


    }
}

File: java/JavaTupleEngine/src/it/mintlab/desktopjava/tupleengine/AtomToken.java
package it.mintlab.desktopjava.tupleengine;

	public class AtomToken implements IToken
	{
		private String atom;
		private Object parameter;
		private boolean withParameter;

		public AtomToken(String atom)
		{
			this.atom = atom;
			this.withParameter = false;
			this.parameter = null;
		}

		public final void setParameter(Object parameter)
		{
			this.parameter = parameter;
			this.withParameter = true;
		}

		public final Object getParameter()
		{
			return parameter;
		}

		@Override
		public String toString()
		{
			return atom;
		}

		public final boolean isWithParameter()
		{
			return this.withParameter;
		}


	}